packages/zero-cache/src/services/mutagen/pusher.test.ts

```diff
@@ -566,6 +566,112 @@ describe('pusher streaming', () => {
       });
     }
   });
+
+  test('fails the stream on ooo mutations', async () => {
+    const fetch = (global.fetch = vi.fn());
+    const oooResponse: PushResponse = {
+      mutations: [
+        {
+          id: {clientID, id: 3},
+          result: {},
+        },
+        {
+          id: {clientID, id: 1},
+          result: {error: 'ooo-mutation'},
+        },
+      ],
+    };
+
+    fetch.mockResolvedValue({
+      ok: true,
+      json: () => Promise.resolve(oooResponse),
+    });
+
+    const pusher = new PusherService(
+      config,
+      lc,
+      'cgid',
+      'http://example.com',
+      'api-key',
+    );
+    void pusher.run();
+
+    const result = pusher.enqueuePush(
+      clientID,
+      wsID,
+      makePush(2, clientID),
+      'jwt',
+    );
+    expect(result.type).toBe('stream');
+
+    if (result.type === 'stream') {
```

Comment:
this is fine but next time you can use assert to do the type narrowing so you can reduce indentation.